restalchemy/storage/sql/dialect/query_builder/q.py

```diff
@@ -141,9 +150,35 @@ def __init__(self, column, session, sort_type=None):
                 raise ValueError("Unknown order: %s" % self._sort_type)
 
     def compile(self):
-        return "%s %s" % (
-            self._session.engine.escape(self._column.name),
-            self._sort_type,
+        column_name = self._column.compile()
+
+        # Handle simple ASC/DESC without NULLS specification
+        if self._sort_type in ("ASC", "DESC"):
+            return "%s %s" % (column_name, self._sort_type)
+
+        # Handle NULLS FIRST/LAST with a generic CASE approach
+        # Works for both PostgreSQL and MySQL (unlike "NULLS FISRT/LAST")
+        order_map = {
+            "ASC NULLS FIRST": ("ASC", 0, 1),
+            "ASC NULLS LAST": ("ASC", 1, 0),
+            "DESC NULLS FIRST": ("DESC", 0, 1),
+            "DESC NULLS LAST": ("DESC", 1, 0),
+        }
+        order, then_, else_ = order_map[self._sort_type]
+        # Like "ORDER BY CASE WHEN table.col IS NULL
```

Comment:
Is there any speed difference for postgres to use `ASC NULLS FIRST` vs `order by case...`?

Reply:
I'd try to check if it's postgres, and if it is - just use _sort_type as is earlier. Maybe I miss something?